src/rulesets/tests/lifecycle.test.ts

```diff
@@ -101,4 +101,65 @@ describe("lifecycle", () => {
       expect(result).toMatchSnapshot();
     });
   });
+
+  describe("sunset", () => {
+    it("fails when the file was removed and not deprecated", async () => {
+      const result = await compare(baseOpenAPI)
+        .to((spec) => {
+          spec.info.title = "Empty";
+          return spec;
+        })
+        .withRule(rules.followSunsetRules, emptyContext);
+      expect(result.results[0].passed).toBeFalsy();
+      expect(result).toMatchSnapshot();
+    });
+
+    describe("sunset schedule", () => {
+      const context: SynkApiCheckContext = {
+        changeDate: "2021-10-20",
+        changeResource: "Example",
+        changeVersion: {
+          date: "2021-10-10",
+          stability: "experimental",
+        },
+        resourceVersions: {
+          Example: {
+            "2021-10-10": {
+              experimental: {
+                deprecatedBy: {
+                  date: "2021-10-20",
+                  stability: "ga",
+                },
+              },
+            },
+          },
+        },
+      };
+
+      it("fails when the schedule isn't met", async () => {
+        const result = await compare(baseOpenAPI)
+          .to((spec) => {
+            spec.info.title = "Empty";
```

Comment:
What if I compare a non-empty OpenAPI spec that happens to have the info.title "Empty"? Is there a more robust way to express this?

Reply:
Per side-channel discussion, a special-use extension would be best here, to make a precise distinction between content and processing metadata. So maybe `spec.extensions['x-optic-ci-empty-spec'] = true` or something to that effect? We can then lay claim to all extensions `x-optic-ci-*`.

While I'll admit an API called "Empty" is really unlikely (or is it... maybe it's a placeholder name used for a generated template OpenAPI, a mock, or "empty.io", the service that stares back when you stare into the void for too long!), for me it's more about providing a precise contract & behavior.

Reply:
I'm all for it, @smizell gave me write permissions on this PR, I'll make the change today

Reply:
waiting on a library release to get this out. We're also bundling some performance fixes for bulk-mode and compare.